docs/source/modules/nn.rst

```diff
@@ -42,6 +42,96 @@ Aggregation Operators
 ---------------------
 
 .. currentmodule:: torch_geometric.nn.aggr
+
+Aggregation functions play an important role in the message passing framework and the readout functions of Graph Neural Networks.
+Specifically, many works in the literature (`Hamilton et al. (2017) <https://arxiv.org/abs/1706.02216>`__, `Xu et al. (2018) <https://arxiv.org/abs/1810.00826>`__, `Corso et al. (2020) <https://arxiv.org/abs/2004.05718>`__, `Li et al. (2020) <https://arxiv.org/abs/2006.07739>`__, `Tailor et al. (2021) <https://arxiv.org/abs/2104.01481>`__) demonstrate that the choice of aggregation functions contributes significantly to the representational power and performance of the model.
+For example, **mean aggregation** captures the distribution (or proportions) of elements, **max aggregation** proves to be advantageous to identify representative elements, and **sum aggregation** enables the learning of structural graph properties (`Xu et al. (2018) <https://arxiv.org/abs/1810.00826>`__).
+Recent works also show that using **multiple aggregations** (`Corso et al. (2020) <https://arxiv.org/abs/2004.05718>`__, `Tailor et al. (2021) <https://arxiv.org/abs/2104.01481>`__) and **learnable aggregations** (`Li et al. (2020) <https://arxiv.org/abs/2006.07739>`__) can potentially gain substantial improvements.
+
+To facilitate further experimentation and unify the concepts of aggregation within GNNs across both :class:`~torch_geometric.nn.conv.MessagePassing` and global readouts, we have made the concept of :class:`~torch_geometric.nn.aggr.Aggregation` a first-class principle in PyG.
+As of now, PyG provides support for various aggregations --- from simple ones (*e.g.*, :obj:`mean`, :obj:`max`, :obj:`sum`), to advanced ones (*e.g.*, :obj:`median`, :obj:`var`, :obj:`std`) and learnable ones (*e.g.*, :obj:`softmax`, :obj:`powermean`), or exotic ones (*e.g.*, :obj:`lstm`, :obj:`sort`):
+
+.. code-block:: python
+
+   from torch_geometric.nn import aggr
+
+   # Simple aggregations:
+   mean_aggr = aggr.MeanAggregation()
+   max_aggr = aggr.MaxAggregation()
+
+   # Advanced aggregations:
+   median_aggr = aggr.MedianAggregation()
+
+   # Learnable aggregations:
+   softmax_aggr = aggr.SoftmaxAggregation(learn=True)
+   powermean_aggr = aggr.PowerMeanAggregation(learn=True)
+
+   # Exotic aggregations:
+   lstm_aggr = aggr.LSTMAggregation(in_channels=..., out_channels=...)
+   sort_aggr = aggr.SortAggregation(k=4)
+
+   x = torch.randn(1000, 64)  # 1000 nodes with 64 features each.
+   index = torch.randint(0, 100, (1000, ))  # Assign nodes to 100 graphs.
+
+   output = mean_aggr(x, index)  #  Output shape: [100, 64]
+
+Notably, all aggregations share the same set of forward arguments, as described in detail in :meth:`torch_geometric.nn.aggr.Aggregation.forward`.
```

Comment:
Do we want to caveat here that there is not yet support for these arguments across all aggregations?

Reply:
Good point. I think we should do this in the doc-strings of the individual modules, using a warning box for example. Let me know if you want to add this in.